torch/distributed/_tensor/device_mesh.py

```diff
@@ -34,7 +34,11 @@
 class _MeshEnv:
     def __init__(self) -> None:
         self.mesh_stack: List[DeviceMesh] = []
+        # child_to_parent_mapping records all the child to parent mesh mapping in the current Mesh Environment
         self.child_to_parent_mapping: Dict[DeviceMesh, DeviceMesh] = {}
+        # parent_mesh_dim_usage records the PT-D API call on a mesh dimension for a given parent_mesh
+        # in the current Mesh Environment
+        self.parent_mesh_dim_usage: Dict[DeviceMesh, Dict[int, str]] = {}
```

Comment:
I feel that we should not record those information to device mesh or any form recording here. My view about this: some integration piece for different parallelism should be enabled only if there's a need for this parallelism. In particular for FSDP + TP case: the extension enablement need to happen in TP API, as TP want to integrate with FSDP/DDP, each parallelism submesh management should happen within their own API